lua/lualine/highlight.lua

```diff
@@ -1,24 +1,58 @@
 local M = {  }
 
-local function highlight (name, foreground, background, special)
-  local command = {
-    'highlight', name,
-    'guifg=' .. foreground,
-    'guibg=' .. background,
-    'gui=' .. (special or 'none'),
-  }
+local function get_cterm_color(color)
+  local function round(num)
+    if num - math.floor(num) < 0.5000000 then
+      return math.floor(num)
+    else
+      return math.ceil(num)
+    end
+  end
+
+  local r = tonumber(color:sub(2,3), 16)
+  local g = tonumber(color:sub(4,5), 16)
+  local b = tonumber(color:sub(6,7), 16)
+  return round(36 * round(r * 5/256) + 
```

Comment:
Could you explain how this works? Maybe then we will be able to figure out a better solution. This creates really weird colors for `gruvbox` theme

Reply:
checkout https://stackoverflow.com/questions/27159322/rgb-values-of-the-colors-in-the-ansi-extended-colors-index-17-255#27165165

The issue is high contrast colors are in first 16 numbers . for 8bit and extended 8bit colors 16-232 are separated as rgb colors after that there are various shades of gray . What I'm doing here is pushing 6byte rgb in that 16-232 range . But we are missing out on those first 16 and later grays . I think thats whats causing them . I'm not sure how to incorparate those colors in this . I might try to add first 16 but I have no idea about later grays . See if you can find a better algorithm

Reply:
tmux has a way of handling grays [here](https://github.com/tmux/tmux/blob/dae2868d1227b95fd076fb4a5efa6256c7245943/colour.c#L57). 
You can look at it closely and let me know if you think their implementation would be an overkill for our usecase or not.

Reply:
I actualy came across that . But didn't realy check what they were doing . I'll check it out when I'm free and let you know

Reply:
Great, I will check it out when I'm free as well and we will be able to proceed further